### PR TITLE
Ros2 implementation branch

### PR DESCRIPTION
**Pull request by bertaveira**
Still a work in progress. As of now tested only for ROS humble and using Ouster data.

Next steps will be to adapt the launch files and parameter files of other Lidars.

### CODE REVIEW

File: src/Modules/Compensator.cpp [hunk 1]
```diff
@@ -69,7 +69,7 @@ extern struct Params Config;
                 upsampled_states (size := imus.size): before t1 and after t2
         */
         States Compensator::upsample(const States& states, const IMUs& imus) {
-            assert (imus.front().time <= states.front().time and states.back().time <= imus.back().time);
+            //assert (imus.front().time <= states.front().time and states.back().time <= imus.back().time);
```
Comment (bertaveira) [inline src/Modules/Compensator.cpp:72]: I had to comment this out. The first part of the assert would always throw an error. I think it has to change from `<=` to `>=` to be as intended. Not sure how it works in the ROS1 version.

Issue created about it #28

**Comment by Huguet57** [comment]
Hi!
What do you use to build it? Colcon? I'm trying to build it with my machine.

**Comment by bertaveira** [comment]
> Hi! What do you use to build it? Colcon? I'm trying to build it with my machine.

Yes I just use `colcon build`

**Comment by Huguet57** [comment]
Hey! It builds and runs! Do you have a sample dataset for Ouster ROS2? Something like 10 seconds would be enough.

**Comment by bertaveira** [comment]
> Huguet57

Yes, I have sent you a bag via email. It drifts with the configuration in this branch. Probably a parameter tuning problem

**Pull request merged**